File: replace_class.py
import os
import re

directory = r"d:\htdocs\CLSU\resources"

# Regex to match whole word 'engineering-page'
pattern = re.compile(r'\bengineering-page\b')

count = 0

for root, dirs, files in os.walk(directory):
    for file in files:
        if file.endswith(('.blade.php', '.css')):
            filepath = os.path.join(root, file)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                if pattern.search(content):
                    new_content = pattern.sub('college-page', content)
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Updated: {filepath}")
                    count += 1
            except Exception as e:
                print(f"Error reading {filepath}: {e}")

print(f"Total files updated: {count}")

File: replace_class.php
<?php
$directory = 'd:/htdocs/CLSU/resources';

$iterator = new RecursiveIteratorIterator(
    new RecursiveDirectoryIterator($directory, RecursiveDirectoryIterator::SKIP_DOTS),
    RecursiveIteratorIterator::SELF_FIRST
);

$count = 0;
foreach ($iterator as $file) {
    if ($file->isFile() && (pathinfo($file->getFilename(), PATHINFO_EXTENSION) === 'php' || pathinfo($file->getFilename(), PATHINFO_EXTENSION) === 'css')) {
        $filepath = $file->getRealPath();
        $content = file_get_contents($filepath);
        if (strpos($content, 'engineering-page') !== false) {
            $new_content = preg_replace('/\bengineering-page\b/', 'college-page', $content);
            file_put_contents($filepath, $new_content);
            echo "Updated: $filepath\n";
            $count++;
        }
    }
}
echo "Total updated: $count\n";
